lib/MusicBrainz/Server/Form/Collection.pm

```diff
@@ -54,6 +54,11 @@ sub options_type_id {
                     $self->ctx->model('CollectionType')->find_by_entity_type($entity_type);
             $types = [grep {$valid_types{$_->{value}}} @$types];
         }
+    } elsif ($collection && $collection->{allowed_entity_type}) {
+        my %valid_types =
+            map { $_->id => 1 }
+                $self->ctx->model('CollectionType')->find_by_entity_type($collection->{allowed_entity_type});
+        $types = [grep {$valid_types{$_->{value}}} @$types];        
```

Comment:
It might be cleaner to do something like `$type_filter = $collection->{allowed_entity_type};` here and `$type_filter = $entity_type;` in the `unless` above so that you can just check if `defined $type_filter` and not repeat this block.